lambda_src/lambda_function.py

```diff
@@ -33,19 +59,22 @@ def async_flow_init(event: Any, context: Any) -> Dict[Text, Any]:
     LOG.debug('Found a destination header and hence using async_flow_init().')
 
     headers = event['headers']
+    req_body = loads(event['body'])
     batch_id = headers[BATCH_ID_HEADER]
     destination = headers[DESTINATION_URI_HEADER]
-    headers.pop(DESTINATION_URI_HEADER)
-    headers['write-uri'] = destination
+    headers['sf-custom-write-uri'] = destination
     lambda_name = context.function_name
     LOG.debug(f'async_flow_init() received destination: {destination}.')
 
     destination_driver = import_module(
         f'geff.drivers.destination_{urlparse(destination).scheme}'
     )
-    # Ignoring style due to dynamic import
-    destination_driver.initialize(destination, batch_id)  # type: ignore
+    # Ignoring style due to dynamic imports
+    for rn, *args in req_body['data']:
+        destination = process_row_params_format(headers, args)['destination_uri']
```

Comment:
`destination_uri` is guaranteed to be there because it's the transformed version of the value in `DESTINATION_URI_HEADER`, and everything else is ignored? that seems like a bit of an indirect coupling. why not just call `format(destination, args)`?

then you can also just inline `process_row_params_format` abstraction, which seems like a tough one to name and explain concisely, anyway.

Reply:
> destination_uri is guaranteed to be there because it's the transformed version of the value in DESTINATION_URI_HEADER, and everything else is ignored? that seems like a bit of an indirect coupling. why not just call format(destination, args)?

Makes sense 👍